hnn_core/pyramidal.py

```diff
@@ -270,23 +298,28 @@ class L2Pyr(Pyr):
         List of dendrites.
     """
 
-    def __init__(self, gid=-1, pos=-1, p={}):
+    def __init__(self, gid=-1, pos=None, geometry=None):
         # Get default L2Pyr params and update them with any
         # corresponding params in p
-        p_all_default = get_L2Pyr_params_default()
-        self.p_all = compare_dictionaries(p_all_default, p)
-
-        # Get somatic, dendritic, and synapse properties
-        p_soma = self._get_soma_props(pos)
+        if geometry is None:
+            geometry = get_L2Pyr_params_default()
+        elif isinstance(geometry, dict):
+            p_all_default = get_L2Pyr_params_default()
+            p_all = compare_dictionaries(p_all_default, params)
 
-        # usage: Pyr.__init__(self, soma_props)
-        Pyr.__init__(self, gid, p_soma)
+            # Get somatic, dendritic, and synapse properties
+            p_soma = self._get_soma_props(pos, p_all)
 
-        p_dend = self._get_dend_props()
-        p_syn = self._get_syn_props()
+            # usage: Pyr.__init__(self, soma_props)
+            Pyr.__init__(self, gid, p_soma)
 
+            p_dend = self._get_dend_props(p_all)
+        elif isinstance(geometry, str):
+            p_dend = _read_geo_cells(geometry)
```

Comment:
Is this still WIP?

Reply:
yep, still WIP. It's a bit of a rabbit-hole ... decided to pause and get some perspectives before continuing!